### ISSUE TITLE
Input group when changing form control size

### ISSUE THREAD

**Issue by daison12006013** [twbs/bootstrap#17810]
```
<div class="input-group">
  <input type="text" class="form-control form-control-sm" placeholder="Message">
  <span class="input-group-btn">
    <button type="button" class="btn btn-primary btn-sm"><i class="fa fa-send"></i> Send</button>
  </span>
</div>
```

The code above shows that there is a bit 1 pixel below on it, even if removing the font awesome, it is still having a pixel.

<img src="http://i.imgur.com/iJMCN6l.png">

The padding size of the button should be ``.275rem`` not ``.25rem`` if we will be basing it to the input size, or applying ``.25rem`` on input. (Vice Versa)

<img src="http://i.imgur.com/UFIamz1.png">
<img src="http://i.imgur.com/Qur0H51.png">

**Comment by cvrebert** [comment]
* What OS and browser are you using?
* Please ensure that your browser's page zoom is at 100% (i.e. not zoomed at all)
* Please re-test against the bleeding-edge `v4-dev` branch.
* Please post your example as a JS Fiddle or JS Bin.

**Comment by daison12006013** [comment]
@cvrebert, I am using chrome...

try this http://jsfiddle.net/L2qetmok/ and try to uncomment the css I did, the property should be like that or vice versa between button and input.

**Comment by twbs-lmvtfy** [comment]
Hi @daison12006013!

You appear to have posted a live example (http://fiddle.jshell.net/L2qetmok/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 38, column 5: [E009](https://github.com/twbs/bootlint/wiki/E009): Button and input sizing within `.input-group`s can cause issues. Instead, use input group sizing classes `.input-group-lg` or `.input-group-sm`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by daison12006013** [comment]
ow, so it should be ``input-group input-group-sm``, damn it should be added in the documentation, or I just missed that line in the docs, lol.

**Comment by cvrebert** [comment]
Actually, that advice might only apply to v3. Will have to check.

**Comment by cvrebert** [comment]
Yup, that advice is still valid in v4.
The docs already show that this is how input group sizing works (http://v4-alpha.getbootstrap.com/components/input-group/#sizing ), but it might be good to add sized `.input-group-btn` examples to that section for clarity.

**Comment by daison12006013** [comment]
@cvrebert, that would probably help others :+1:

**Comment by mdo** [comment]
This was addressed awhile back I believe, as part of #19141. Seeing no issues in the linked jsfiddle with the latest CSS applied.
